mesheryctl/internal/cli/root/perf/apply_test.go

```diff
@@ -0,0 +1,268 @@
+package perf
+
+import (
+	"bytes"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/jarcoal/httpmock"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+)
+
+func TestApplyCmd(t *testing.T) {
+	// setup current context
+	utils.SetupContextEnv(t)
+
+	// initialize mock server for handling requests
+	utils.StartMockery(t)
+
+	// create a test helper
+	testContext := utils.NewTestHelper(t)
+
+	// get current directory
+	_, filename, _, ok := runtime.Caller(0)
+	if !ok {
+		t.Fatal("problems recovering caller information")
+	}
+	currDir := filepath.Dir(filename)
+	fixturesDir := filepath.Join(currDir, "fixtures")
+
+	// test scenrios for fetching data
+	tests := []struct {
+		Name             string
+		Args             []string
+		ExpectedResponse string
+		URLs             []utils.MockURL
+		Token            string
+		ExpectError      bool
+	}{
+		{
+			Name:             "Run Test with Existing profile",
+			Args:             []string{"apply", "new", "--token", filepath.Join(fixturesDir, "token.golden")},
+			ExpectedResponse: "apply.success.output.golden",
+			URLs: []utils.MockURL{
+				{
+					Method:       "GET",
+					URL:          testContext.BaseURL + "/api/user/performance/profiles?search=new",
+					Response:     "apply.fetch.profile.response.golden",
+					ResponseCode: 200,
+				},
+				{
+					Method:       "GET",
+					URL:          testContext.BaseURL + "/api/user/performance/profiles/" + "8f3daf25-e58e-4c59-8bf8-f474b76463ec" + "/run",
```

Comment:
can we define these uuids as constants ? For e.g. ValidPerfProfileID or ExistingPerfProfileID, NonExistingPerfProfileID

Reply:
Nice, thought. I implemented this in the latest commit.